src/Sylius/Behat/Context/Setup/PromotionContext.php

```diff
@@ -142,24 +138,20 @@ public function itGivesFixedDiscountToEveryOrder(PromotionInterface $promotion,
      */
     public function itGivesPercentageDiscountToEveryOrder(PromotionInterface $promotion, $discount)
     {
-        $action = $this->actionFactory->createPercentageDiscount($discount);
-        $promotion->addAction($action);
-
-        $this->objectManager->flush();
+        $this->createPercentagePromotion($promotion, $discount);
     }
 
     /**
      * @Given /^([^"]+) gives ("[^"]+") fixed discount to every order with quantity at least ([^"]+)$/
```

Comment:
What do you think about changing the first `([^"]+)` into `(it|this promotion)`?